walk-in-interview/src/test/java/com/google/job/data/JobsDatabaseTest.java

```diff
@@ -0,0 +1,107 @@
+package com.google.job.data;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.firestore.FireStoreUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests for {@link JobsDatabase} class. */
+public final class JobsDatabaseTest {
+    // TODO(issue/15): Add failure test case
+
+    private static final String TEST_JOB_COLLECTION = "Jobs";
+    private static final int BATCH_SIZE = 10;
+
+    Firestore firestore;
+    JobsDatabase jobsDatabase;
+
+    @Before
+    public void setUp() throws IOException {
+        firestore = FireStoreUtils.getFireStore();
+        jobsDatabase = new JobsDatabase(firestore);
+    }
+
+    @Test
+    public void addJob_NormalInput_success() throws ExecutionException, InterruptedException {
+        String expectedJobName = "Software Engineer";
+        Job job = new Job(expectedJobName);
+        Future<DocumentReference> addedJobFuture = jobsDatabase.addJob(job);
+
+        DocumentReference documentReference = addedJobFuture.get();
+        // Asynchronously retrieve the document.
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+
+        // future.get() blocks on response.
+        DocumentSnapshot document = future.get();
+        String jobId = document.getId();
+
+        Job actualJob = document.toObject(Job.class);
+        String actualJobName = actualJob.getJobTitle();
+
+        assertEquals(expectedJobName, actualJobName);
+
+        firestore.collection(TEST_JOB_COLLECTION).document(jobId).delete();
```

Comment:
i recommend you don't do that here. if your test fails yoru collection will be corrupted and the next test that runs after this will start from a corrupted collection. Instead you can add either in `@Before` or `@After` some cleanup where you drop the collection.